### ISSUE TITLE
Upgrade Process Stalled at 0% under Privacy Mode

### ISSUE THREAD

**Issue by mehul0810** [WordImpress/Give#2697]
## Issue Overview
This is an issue which i have reproduce on Local by Flywheel as well as FlyWheel Demo Site. Both the instance are in privacy mode. I have checked it with `update-process` branch which has all the latest 2.0.1 changes.

Possible cause all the issues we have received somehow relates to `admin-ajax.php` file which is secured with http authentication by the end user as a part of security concern.

## Expected Behavior
Upgrade process should run smoothly.

## Current Behavior
Upgrade Process is stalled.

## Steps to Reproduce (for bugs)
1. Create a WP Instance with HTTP Proxy Authentication Enabled on admin-ajax.php or whole site. For more reference: https://www.digitalocean.com/community/tutorials/how-to-set-up-basic-http-authentication-with-nginx-on-ubuntu-14-04
2. Install Give 1.8.19 
3. Create couple of forms and donations
4. Upgrade to Give 2.0.1 
5. You will notice the upgrade process stalled. Also, try pausing and restarting update process.

## Related Issues and/or PRs
https://github.com/WordImpress/Give/issues/2671

## Todos
- [ ] Tests
- [ ] Documentation

## WordPress Environment
<details>

### WordPress Environment ###

Home URL: http://givetestdemo.flywheelsites.com
Site URL: http://givetestdemo.flywheelsites.com
WP Version: 4.9.2
WP Multisite: –
WP Memory Limit: 128 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
Permalink Structure: 
Show on Front: posts
Table Prefix Length: wp_llhwrtc8ec_
Table Prefix Length: 14
Table Prefix Status: Acceptable
Admin AJAX: Accessible
Registered Post Statuses: publish, future, draft, pending, private, trash, auto-draft, inherit, refunded, failed, revoked, cancelled, abandoned, processing, preapproval

### Server Environment ###

Hosting Provider: DBH: 127.0.0.1, SRV: givetestdemo.flywheelsites.com
TLS Connection: Connection uses TLS 1.2
TLS Connection: Probably Okay
Server Info: Flywheel/4.1.0
PHP Version: 7.1.12-2+ubuntu16.04.1+deb.sury.org+2
PHP Post Max Size: 300 MB
PHP Time Limit: 30
PHP Max Input Vars: 4000
PHP Max Upload Size: 300 MB
cURL Version: 7.47.0, OpenSSL/1.0.2g
SUHOSIN Installed: –
MySQL Version: 5.6.32
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
gzip: ✔
GD Graphics Library: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Give Configuration ###

Give Version: 2.0.1
Give Cache: Enabled
Database Updates: 9 of 20 updates still need to run.
Give Cache: Enabled
Give Cache: ✔New Donation✔Donation Receipt✔New Offline Donation✔Offline Donation Instructions✔New User Registration✔User Registration Information✔Email access
Upgraded From: 2.0.1
Test Mode: Enabled
Currency Code: USD
Currency Position: Before
Decimal Separator: .
Thousands Separator: ,
Success Page: http://givetestdemo.flywheelsites.com/?page_id=5
Failure Page: http://givetestdemo.flywheelsites.com/?page_id=6
Donation History Page: http://givetestdemo.flywheelsites.com/?page_id=7
Give Forms Slug: /donations/
Enabled Payment Gateways: Test Donation, Offline Donation
Default Payment Gateway: Test Donation
PayPal IPN Verification: Enabled
PayPal IPN Notifications: N/A
Donor Email Access: Enabled

### Session Configuration ###

Give Use Sessions: Enabled
Session: Disabled

### Active Give Add-ons ###

Give - Donation Plugin: ❌ Unlicensed – by WordImpress – 2.0.1

### Other Active Plugins ###

Query Monitor: by John Blackbourn – 2.17.0
WP Statistics: by Verona Labs – 12.3.5

### Theme ###

Name: Twenty Seventeen
Version: 1.2
Author URL: https://wordpress.org/
Child Theme: No – If you're modifying Give on a parent theme you didn't build personally, then we recommend using a child theme. See: How to Create a Child Theme

</details>

**Comment by ravinderk** [comment]
@mehul0810 In ajax testing function (`give_test_ajax_works`) we will return `true` if there are HTTP Authentication, to prevent showing update ajax notice. But We are not able to access `admin-ajax.php` because before hitting update trigger URL server protect it. Db update will run by cron but you will not able to see immediately effect ( 5 min cron time ).

@DevinWalker Rather than return `true` from `give_test_ajax_works` on `401` we can update message with blog post link https://givewp.com/documentation/core/frequent-troubleshooting-, so admin can debug more.

@DevinWalker @mehul0810 Let me know what do you think?

**Comment by mehul0810** [comment]
@ravinderk I agree with your suggestion. We should update the notice on `401` rather than allowing access to run the upgrade routine because when admin-ajax.php is blocked using privacy mode then upgrade routine will get affected. 

Hence, we need to guide end users to follow the troubleshooting steps while running upgrade scripts. Also, the same point regarding admin-ajax blocking issue is mentioned in WordPress documentation https://codex.wordpress.org/Brute_Force_Attacks

**Comment by DevinWalker** [comment]
Actually, I didn't change this message because the outcome was confusing as to what the functionality is and the expected message.
